content/en/docs/tasks/configure-pod-container/configure-gmsa.md

```diff
@@ -0,0 +1,302 @@
+
+### Create GMSA credspec resources
+With the GMSACredentialSpec CRD installed, GMSA credspec custom resources can now be configured. The GMSA credential spec does not contain secret or sensitive data. It is information that a container runtime can use to describe the desired GMSA of a container to Windows. The GMSA credspec resources can be generated in JSON format with a utility [PowerShell][] script. Following are the steps for generating a GMSA credspec YAML based on the JSON:
+1. Import the CredentialSpec module: `ipmo CredentialSpec.psm1`
+2. Create a credential spec in JSON format using `New-CredentialSpec`. To create a GMSA credspec named WebApp1: `New-CredentialSpec -Name WebApp1 -AccountName WebApp1 -Domain $(Get-ADDomain -Current LocalComputer)`
+3. Use `Get-CredentialSpec` to show the path of the JSON file. 
+4. Convert the credspec file from JSON to YAML format and apply the necessary header fields `apiVersion`, `kind`, `metadata` and `credspec` to make it a GMSACredentialSpec custom resource. An example based on a GMSA credspec named WebApp1 is below:
+
+```
+apiVersion: windows.k8s.io/v1alpha1
+kind: GMSACredentialSpec
+metadata:
+  name: gmsa-WebApp1  #This is an arbitrary name but it will be used as a reference
+credspec:
+  ActiveDirectoryConfig:
+    GroupManagedServiceAccounts:
+    - Name: WebApp1   #Username of the GMSA account
+      Scope: CONTOSO  #NETBIOS Domain Name
+    - Name: WebApp1   #Username of the GMSA account
+      Scope: contoso.com #DNS Domain Name
+  CmsPlugins:
+  - ActiveDirectory
+  DomainJoinConfig:
+    DnsName: contoso.com  #DNS Domain Name
+    DnsTreeName: contoso.com #DNS Domain Name Root
+    Guid: 244818ae-87ac-4fcd-92ec-e79e5252348a  #GUID
+    MachineAccountName: WebApp1 #Username of the GMSA account
+    NetBiosName: CONTOSO  #NETBIOS Domain Name
+    Sid: S-1-5-21-2126449477-2524075714-3094792973 #SID of GMSA
+```
+
+5. Deploy the credential spec resource: `kubectl apply -f gmsa-Webapp1-credspec.yml`
+
+[PowerShell]: https://github.com/MicrosoftDocs/Virtualization-Documentation/blob/live/windows-server-container-tools/ServiceAccounts/CredentialSpec.psm1
+
+### Install Webhooks for GMSA
+Two webhooks need to be configured on the Kubernetes cluster to populate and validate GMSA credential spec references at the pod or container level. The mutating webhook expands references to GMSAs (by name from a pod specification) into the full credential spec in JSON form within the pod spec. The validating webhook ensures all references to GMSAs are authorized to be used by the pod service account. 
+
+Installing the webhooks require several steps:
+1. Create a certificate key pair (that will be used to allow the webhook container to communicate to the cluster)
+2. Install a secret with the certificate from above.
+3. Create a deployment for the core webhook logic. 
+4. Create the validating and mutating webhook configurations referring to the deployment. 
```

Comment:
Could be worth mentioning that they can choose to do all that with a single command, as explained briefly in https://github.com/kubernetes-sigs/windows-gmsa/tree/master/admission-webhook#how-to-deploy (and that the `--dry-run` flag in that script would let them examine the k8s manifests before applying any chance); but that they can also choose to do this manually with the steps below?